S2VX.Game/Story/Note/Notes.cs

```diff
@@ -36,6 +37,23 @@ public class Notes : CompositeDrawable {
             RemoveInternal(note);
         }
 
+        // Before starting the Story in the PlayScreen, we want to explicitly
+        // remove GameNotes up to some certain track time. This is so that we
+        // won't hear Miss hitsounds and prematurely calculate score.
+        public void RemoveNotesUpTo(double trackTime) {
+            int validIndex;
+            for (validIndex = 0; validIndex < Children.Count; ++validIndex) {
+                if (Children[validIndex].EndTime < trackTime) {
+                    break;
+                }
+            }
+
+            var newNotes = Children.Take(validIndex).ToList();
+            Children = newNotes;
+            ClearInternal(false);
```

Comment:
Do we need the old notes list? What does disposeChildren = false do for us here?

Reply:
We need to call ClearInternal(false) so that Notes are not disposed of before reuse. The main problem lies with the next line: InternalChildren = Childre`. When you assign InternalChildren a new Drawable list, it first invalidates all the existing Drawables before setting the new Drawables. This is a problem for `RemoveNotesUpTo` because some of the same Notes are being used before and after InternalChildren is set.

We need to call ClearInternal(false) so that we first clear the InternalChildren without invalidating the Drawables, then we can safely set InternalChildren without invalidation errors.